src/main/java/org/jabref/gui/preferences/customentrytypes/CustomEntryTypesTab.java

```diff
@@ -143,6 +146,29 @@ private void setupEntryTypesTable() {
 
     private void setupFieldsTable() {
         fieldNameColumn.setCellValueFactory(item -> item.getValue().nameProperty());
+        fieldNameColumn.setCellFactory(TextFieldTableCell.forTableColumn());
+        fieldNameColumn.setEditable(true);
+        fieldNameColumn.setOnEditCommit(
+                (TableColumn.CellEditEvent<FieldViewModel, String> event) -> {
+                    String newFieldValue = event.getNewValue();
+                    UnknownField field = (UnknownField) event.getRowValue().getField();
+                    EntryTypeViewModel selectedEntryType = viewModel.selectedEntryTypeProperty().get();
+                    ObservableList<FieldViewModel> entryFields = selectedEntryType.fields();
+
+                    boolean fieldExists = entryFields.stream().anyMatch(fieldViewModel ->
+                            fieldViewModel.nameProperty().getValue().equals(newFieldValue));
+
+                    if (!fieldExists) {
+                        event.getRowValue().setField(newFieldValue);
+                        field.setName(newFieldValue);
+                    } else {
+                        dialogService.showWarningDialogAndWait(
+                                Localization.lang("Duplicate fields"),
+                                Localization.lang("Warning: You added field \"%0\" twice. Only one will be kept.", newFieldValue));
+                        event.getTableView().edit(-1, null);
+                        event.getTableView().refresh();
+                    }
```

Comment:
Modern software seldom makes use of popups. I would use the `notifiy` functionality. The user has no choice. Moreover, the current code keeps the old casing. Thus, instead of `newFieldValue`, the existing one should be output. As a consequence, the code of `fieldExists` needs to be modified accordingly.

Reply:
The reason I use `newFieldValue` instead of the existing one is to see if the user input value (`newFieldValue`) exist in the current fields. Let's say we already have two new Field, **Name** and **Test**. When a user try to edit **Name** to **Test**, he would double click the **Name** Field and type "Test", so "Test" is the newFieldValue. So I think we have to see what user input is and compare it with the existing Field. In this case, **Test** already exist, so we will notify the user.

If we choose the existing one by using the method `field.getDisplayName()`, the value would be **Name** since we haven't modify the field yet. In this case, we always get the `fieldExists` value to be True

Let me know if I understand the required change correctly :)

Reply:
Sorry that I was not clear in my terms. Thank you for explaining a scenario. I hope, this short explanation based on your scenario helps:

I was trying to say: User renames to **TEST**, but **Test** exists. I assume the user knows, he input **TEST**, but forgot about the existing field **Test**. Therefore, **Test** should be displayed and not **TEST**.